ivy/functional/frontends/jax/nn/non_linear_activations.py

```diff
@@ -271,10 +271,13 @@ def soft_sign(x):
     return ret.astype(dtype)
 
 
-def softmax(x, /, *, axis=-1):
-    dtype = _type_conversion(x).dtype
+def softmax(x, axis=-1, where=None, initial=None):
     ret = ivy.softmax(x, axis=axis)
-    return ret.astype(dtype)
+    if initial:
+        if ivy.is_array(where):
+            x = ivy.where(where, x, x.full_like(initial))
```

Comment:
I've tried it out, for now the case when there is a combination of `where` and `initial` results in the error you pointed out, I'll have to dig in a bit more to know why, but you can fix the other first and maybe comment out the `where` argument and its related code before the issue is fixed :)